girder_volview/__init__.py

```diff
@@ -137,10 +140,19 @@ def volViewLoadableFolder(self, folder):
     return {"loadable": loadable}
 
 
-def uploadSession(model, parentId, user, size):
+def uploadSession(model, parentId, user, size, metadata=None):
     # modified from girder.api.v1.file.File.initUpload
     parentType = model.__name__.lower()
-    name = "session.volview.zip"
+    name = f"session{SESSION_ZIP_EXTENSION}"
+    try:
+        # If the session was created due to a filter, include that as part of
+        # the name.  Don't fail if it isn't formated as we expect.
+        if metadata and metadata.get('linkedResources') and metadata['linkedResources'].get('filter'):
+            firstFilter = list(iter(metadata['linkedResources']['filter'].values()))[0]
```

Comment:
When do we expect the formating to not be as we expect?  Incase large_image changes the shape of `filter` down the road?

Reply:
This is my belt-and-suspenders guard; I _think_ it could do without the guard, but since it is based on metadata that _can_ be changed by a user, I don't really trust it.